协议H2,版本5.7.0，Server deserialize error，具体堆栈错误如下，之前协议时bolt时没有错误

com.alipay.sofa.rpc.core.exception.SofaRpcException: com.alipay.sofa.rpc.core.exception.SofaRpcException: Server deserialize error, readObject: unknown code 0x59 Y
	at com.alipay.sofa.rpc.transport.http.AbstractHttp2ClientTransport.syncSend(AbstractHttp2ClientTransport.java:258)
	at com.alipay.sofa.rpc.client.AbstractCluster.doSendMsg(AbstractCluster.java:535)
	at com.alipay.sofa.rpc.client.AbstractCluster.sendMsg(AbstractCluster.java:506)
	at com.alipay.sofa.rpc.filter.ConsumerInvoker.invoke(ConsumerInvoker.java:60)
	at com.alipay.sofa.rpc.filter.sofatracer.ConsumerTracerFilter.invoke(ConsumerTracerFilter.java:66)
	at com.alipay.sofa.rpc.filter.FilterInvoker.invoke(FilterInvoker.java:98)
	at com.alipay.sofa.rpc.filter.RpcReferenceContextFilter.invoke(RpcReferenceContextFilter.java:80)
	at com.alipay.sofa.rpc.filter.FilterInvoker.invoke(FilterInvoker.java:98)
	at com.alipay.sofa.rpc.filter.ConsumerExceptionFilter.invoke(ConsumerExceptionFilter.java:37)
	at com.alipay.sofa.rpc.filter.FilterInvoker.invoke(FilterInvoker.java:98)
	at com.alipay.sofa.rpc.filter.FilterChain.invoke(FilterChain.java:262)
	at com.alipay.sofa.rpc.client.AbstractCluster.filterChain(AbstractCluster.java:499)
	at com.alipay.sofa.rpc.client.FailoverCluster.doInvoke(FailoverCluster.java:67)
	at com.alipay.sofa.rpc.client.AbstractCluster.invoke(AbstractCluster.java:288)
	at com.alipay.sofa.rpc.client.ClientProxyInvoker.invoke(ClientProxyInvoker.java:83)
	at com.jinninghui.newspiral.gateway.ServiceForPeer_proxy_1.addGenericMsg(ServiceForPeer_proxy_1.java)
	at com.jinninghui.newspiral.p2p.impl.hotstuff.ServiceForPeerClient.addGenericMsg(ServiceForPeerClient.java:187)
	at com.jinninghui.newspiral.p2p.impl.hotstuff.ChannelClientMgr.broadcastGenericMsg(ChannelClientMgr.java:105)
	at com.jinninghui.newspiral.p2p.impl.hotstuff.HotStuffRPC.broadcastGenericMsg(HotStuffRPC.java:29)
	at sun.reflect.GeneratedMethodAccessor216.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.alipay.sofa.runtime.service.binding.JvmBindingAdapter$JvmServiceInvoker.doInvoke(JvmBindingAdapter.java:200)
	at com.alipay.sofa.runtime.spi.service.ServiceProxy.invoke(ServiceProxy.java:39)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212)
	at com.sun.proxy.$Proxy99.broadcastGenericMsg(Unknown Source)
	at com.jinninghui.newspiral.consensus.impl.hotstuff.NewSpiralHotStuffProcessor.tryCreateBlockAndBroadcast(NewSpiralHotStuffProcessor.java:624)
	at com.jinninghui.newspiral.consensus.impl.hotstuff.NewSpiralHotStuffProcessor.runOnceAsConsensusPeer(NewSpiralHotStuffProcessor.java:183)
	at com.jinninghui.newspiral.consensus.impl.hotstuff.NewSpiralHotStuffProcessor.run(NewSpiralHotStuffProcessor.java:98)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.util.concurrent.ExecutionException: com.alipay.sofa.rpc.core.exception.SofaRpcException: Server deserialize error, readObject: unknown code 0x59 Y
	at com.alipay.sofa.rpc.message.http.HttpResponseFuture.getNowResponse(HttpResponseFuture.java:158)
	at com.alipay.sofa.rpc.message.http.HttpResponseFuture.getSofaResponse(HttpResponseFuture.java:146)
	at com.alipay.sofa.rpc.transport.http.AbstractHttp2ClientTransport.doInvokeSync(AbstractHttp2ClientTransport.java:283)
	at com.alipay.sofa.rpc.transport.http.AbstractHttp2ClientTransport.syncSend(AbstractHttp2ClientTransport.java:252)
	... 30 more
Caused by: com.alipay.sofa.rpc.core.exception.SofaRpcException: Server deserialize error, readObject: unknown code 0x59 Y
	at com.alipay.sofa.rpc.transport.http.AbstractHttpClientHandler.receiveHttpResponse(AbstractHttpClientHandler.java:173)
	at com.alipay.sofa.rpc.transport.http.Http2ClientChannelHandler.channelRead0(Http2ClientChannelHandler.java:99)
	at com.alipay.sofa.rpc.transport.http.Http2ClientChannelHandler.channelRead0(Http2ClientChannelHandler.java:46)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.http2.InboundHttp2ToHttpAdapter.fireChannelRead(InboundHttp2ToHttpAdapter.java:138)
	at io.netty.handler.codec.http2.InboundHttp2ToHttpAdapter.onDataRead(InboundHttp2ToHttpAdapter.java:249)
	at io.netty.handler.codec.http2.DelegatingDecompressorFrameListener.onDataRead(DelegatingDecompressorFrameListener.java:78)
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onDataRead(DefaultHttp2ConnectionDecoder.java:236)
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readDataFrame(DefaultHttp2FrameReader.java:421)
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.processPayloadState(DefaultHttp2FrameReader.java:251)
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readFrame(DefaultHttp2FrameReader.java:160)
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder.decodeFrame(DefaultHttp2ConnectionDecoder.java:118)
	at io.netty.handler.codec.http2.Http2ConnectionHandler$FrameDecoder.decode(Http2ConnectionHandler.java:390)
	at io.netty.handler.codec.http2.Http2ConnectionHandler.decode(Http2ConnectionHandler.java:450)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1429)
	at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1211)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1245)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:628)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:563)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:480)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:442)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	... 1 more

--- a/remoting/remoting-http/src/main/java/com/alipay/sofa/rpc/transport/http/Http2ServerChannelHandler.java
+++ b/remoting/remoting-http/src/main/java/com/alipay/sofa/rpc/transport/http/Http2ServerChannelHandler.java
@@ -63,6 +63,7 @@ public final class Http2ServerChannelHandler extends Http2ConnectionHandler impl
                                                                    .getLogger(Http2ServerChannelHandler.class);
 
     private final Http2Connection.PropertyKey headerKey        = encoder().connection().newKey();
+    private final Http2Connection.PropertyKey messageKey       = encoder().connection().newKey();
 
     private final HttpServerHandler           serverHandler;
 
@@ -114,11 +115,20 @@ public final class Http2ServerChannelHandler extends Http2ConnectionHandler impl
     @Override
     public int onDataRead(ChannelHandlerContext ctx, int streamId, ByteBuf data, int padding, boolean endOfStream) {
         int processed = data.readableBytes() + padding;
+
+        Http2Stream http2Stream = connection().stream(streamId);
+        ByteBuf msg = http2Stream.getProperty(messageKey);
+        if (msg == null) {
+            msg = ctx.alloc().buffer();
+            http2Stream.setProperty(messageKey, msg);
+        }
+        final int dataReadableBytes = data.readableBytes();
+        msg.writeBytes(data, data.readerIndex(), dataReadableBytes);
+
         if (endOfStream) {
-            Http2Stream http2Stream = connection().stream(streamId);
             // read cached http2 header from stream
             Http2Headers headers = http2Stream.getProperty(headerKey);
-            handleRequest(ctx, streamId, headers, data);
+            handleRequest(ctx, streamId, headers, msg);
         }
         return processed;
     }

